src/main/java/com/consolefire/orm/ContextAutowireFactory.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Created at 29-Apr-2016 by sabuj.das
+ */
+package com.consolefire.orm;
+
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author sabuj.das
+ *
+ */
+public final class ContextAutowireFactory implements ApplicationContextAware {
+
+    private static ContextAutowireFactory instance;
+    private ApplicationContext applicationContext;
+
+    /**
+     * 
+     */
+    private ContextAutowireFactory() {
+
+    }
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    /**
+     * @return the factory instance
+     */
+    public static ContextAutowireFactory getInstance() {
+        if(null == instance) {
+            synchronized (ContextAutowireFactory.class) {
```

Comment:
@sabuj-das there is chances that this will return the un constructed object, You can avoid this by marking instance as volatile or some volatile flag to make sure changes are written to main memory.